cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -0,0 +1,279 @@
+
+template <typename Task = task::by_default>
+class descriptor_impl;
+
+template <typename Task = task::by_default>
+class model_impl;
+} // namespace detail
+
+namespace method {
+struct dense {};
+struct hist {};
+using by_default = dense;
+} // namespace method
+
+enum class variable_importance_mode {
+    none, /* Do not compute */
+    mdi, /* Mean Decrease Impurity.
+                       Computed as the sum of weighted impurity decreases for all nodes where the variable is used,
+                       averaged over all trees in the forest */
+    mda_raw, /* Mean Decrease Accuracy (permutation importance).
+                       For each tree, the prediction error on the out-of-bag portion of the data is computed
+                       (error rate for classification, MSE for regression).
+                       The same is done after permuting each predictor variable.
+                       The difference between the two are then averaged over all trees. */
+    mda_scaled /* Mean Decrease Accuracy (permutation importance).
+                       This is MDA_Raw value scaled by its standard deviation. */
+};
+
+enum class train_result_to_compute {
+    compute_out_of_bag_error                 = 0x00000001ULL,
+    compute_out_of_bag_error_per_observation = 0x00000002ULL
+};
+
+enum class infer_result_to_compute {
+    compute_class_labels =
+        0x00000001ULL, /*!< Numeric table of size n x 1 with the predicted labels >*/
+    compute_class_probabilities =
+        0x00000002ULL /*!< Numeric table of size n x p with the predicted class probabilities for each observation >*/
+};
+
+enum class voting_method { weighted, unweighted };
+
+template <typename Task = task::by_default>
+class descriptor_base : public base {
+    friend dal::detail::pimpl_accessor;
+
+public:
+    using tag_t    = detail::tag;
+    using float_t  = float;
+    using task_t   = Task;
+    using method_t = method::by_default;
+    using pimpl    = typename dal::detail::pimpl<detail::descriptor_impl<task_t>>;
+    template <typename T>
+    using is_classification_t =
+        std::enable_if_t<std::is_same_v<T, std::decay_t<task::classification>>>;
+
+    descriptor_base();
+
+    //engines::EnginePtr engine;             /*!< Engine for the random numbers generator used by the algorithms */
+
+    double get_observations_per_tree_fraction() const;
+    double get_impurity_threshold() const;
+    double get_min_weight_fraction_in_leaf_node() const;
+    double get_min_impurity_decrease_in_split_node() const;
+
+    std::int64_t get_tree_count() const;
+    std::int64_t get_features_per_node() const;
+    std::int64_t get_max_tree_depth() const;
+    std::int64_t get_min_observations_in_leaf_node() const;
+    std::int64_t get_min_observations_in_split_node() const;
+    std::int64_t get_max_leaf_nodes() const;
+
+    bool get_memory_saving_mode() const;
+    bool get_bootstrap() const;
+
+    std::uint64_t get_train_results_to_compute() const;
```

Comment:
As I understand it, the  descriptor should be independent of type tasks, step algorithm and etc. Why not just leave:
```suggestion
    std::uint64_t get_results_to_compute() const;
```